ferrocene/ci/scripts/persist-between-jobs.sh

```diff
@@ -47,10 +54,19 @@ case "$1" in
             exit 1
         fi
 
-        # Call `zstd` separately to be able to use all cores available (`-T0`)
-        # and the lowest compression level possible, to speed the compression
-        # as much as possible.
-        tar c --exclude build/metrics.json $@ | zstd -1 -T0 | aws s3 cp - "$(s3_url "${CIRCLE_JOB}")"
+        if [[ "${OSTYPE}" = "msys" ]]; then
+            # CircleCI's folder path structure is too long to support these links on unpack, so we ignore them.
+            "${TAR}" -cf- --exclude build/metrics.json \
+                --exclude build/x86_64-pc-windows-msvc/stage0-sysroot/lib/rustlib/rustc-src/rust --exclude build/x86_64-pc-windows-msvc/stage0-sysroot/lib/rustlib/src/rust --exclude build/x86_64-pc-windows-msvc/stage1/lib/rustlib/rustc-src/rust --exclude build/x86_64-pc-windows-msvc/stage1/lib/rustlib/src/rust --exclude build/x86_64-pc-windows-msvc/stage2/lib/rustlib/rustc-src/rust --exclude build/x86_64-pc-windows-msvc/stage2/lib/rustlib/src/rust --exclude build/host \
+                --preserve-permissions --format=posix $@ \
+                | zstd -1 -T0 \
+                | aws s3 cp - "$(s3_url "${CIRCLE_JOB}")"
+        else
+            # Call `zstd` separately to be able to use all cores available (`-T0`)
+            # and the lowest compression level possible, to speed the compression
+            # as much as possible.
+            "${TAR}" c --exclude build/metrics.json $@ | zstd -1 -T0 | aws s3 cp - "$(s3_url "${CIRCLE_JOB}")"
+        fi
```

Comment:
Can we keep the underlying command the same? `-cf-` vs `c` should be equivalent, and the `--exclude` could be added as a `tar_flags` variable.

Reply:
I initially had this but had some issues with shell expansion. I'll revisit it.

Reply:
Wait, we're not even calling this. I'm just gonna revert.